dev/lint-commits.sh

```diff
@@ -19,14 +19,20 @@ fi
 BASE_COMMIT="$1"
 HEAD_COMMIT="$2"
 
-# git diff --check
-# uses .gitattributes to know what to check
-if git diff --check "$BASE_COMMIT" "$HEAD_COMMIT";
+bad=()
+while IFS= read -r commit; do
+    # git diff --check
+    # uses .gitattributes to know what to check
+    if ! git diff --check "${commit}~" "$commit";
```

Comment:
I'm on git `2.11.0` and `<rev>~` is not documented, while `<rev>^` is (it seems `~` has to be followed by a number, here I guess you meant  `1`).  Conversely `^` can be used alone and means `^1`.